전남대_12조_픽업셔틀_4주차

rktdnjs opened the pull request:
## PR 타입(하나 이상의 PR 타입을 선택해주세요)

- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [x] CSS등 UI 수정
- [x] 의존성, 환경 변수, 빌드 관련 코드 업데이트

## 작업 사항

- **프로젝트 사전 세팅**
- `Atom Components` 개발
  - `Button, Card, Footer, Input, Label, Loader, Nav, SelectInput, SubLabel`
- `Molecules Components` 개발
  - `Labels`
- 카카오 소셜 로그인 페이지 개발 진행
  - 인가 코드 전달하는 과정에서 로더 적용
- 공고 작성 입장 페이지 개발 진행
- 홈페이지 개발 진행
  - `Nav, Footer` 적용
- 전체적인 레이아웃 및 CSS 적용 & 일부 eslint 규칙 수정
- 4주차는 기능보다는 전체적인 틀을 잡고 사용할 Atom 컴포넌트를 구현하는데 집중하였습니다.

## ETC

안녕하십니까 성중원 멘토님! 이번 주차부터 멘토링을 함께 하게 된 12조 픽업셔틀입니다.
앞으로의 멘토링 잘 부탁드리며, 이번주차 4Week와 관련된 질문 사항은 아래 링크에 따로 남겨놓았습니다!
감사합니다.😊

> 질문 사항을 정리한 링크입니다! (멘토 피드백 일지)

https://www.notion.so/FE-5-0811f117cc324e84bc4093d802530407

rktdnjs committed:
[conf] : 프로젝트 사전 세팅

rktdnjs committed:
[conf] : 프로젝트 사전 세팅 - Prettier & eslint 수정

rktdnjs committed:
[docs] : Update README.md - 커밋 & 코딩 컨벤션 추가

rktdnjs committed:
[conf] : Pretendard 폰트 전역 적용 & 기본 레이아웃

- 현재 잡아놓은 레이아웃은 iframe을 이용한 레이아웃 x

baegyeong committed:
[conf]: eslint prop-types off 지정

baegyeong committed:
[conf]: props spreading 허용

baegyeong committed:
[feat]: Input 컴포넌트 구현 (#4)

baegyeong committed:
[conf]: eslint label control 지정

baegyeong committed:
[feat]: label 컴포넌트 구현

baegyeong committed:
[feat]: SubLabel 컴포넌트 구현

- 기본 라벨 아래에 도움말 라벨 역할

rktdnjs committed:
[conf] : eslint prop-types & props-no-spreading off

rktdnjs committed:
[conf] : eslint button-has-types off

rktdnjs committed:
[feat] : Button.jsx 구현

rktdnjs committed:
[feat] : Card.jsx 구현 & comma, date 함수 구현

- comma : num 데이터를 1,000과 같은 형태로 변경해주는 함수
- date : unix timestamp 값을 date의 형태로 변경해주는 함수

baegyeong committed:
[style]: blue color 지정

baegyeong committed:
[conf]: eslint 절대 경로 인식 지정

baegyeong committed:
[feat]: 공고입장 페이지 라우터 지정

baegyeong committed:
[feat]: 공고 작성 페이지 구현

kimdanbin committed:
[feat]:Footer 구현

baegyeong committed:
[style]: label 컴포넌트 block 지정

baegyeong committed:
[style]: SubLabel, Input block 및 텍스트 크기 지정

baegyeong committed:
[feat]: 공고 작성 페이지 라우터 지정

baegyeong committed:
Merge branch 'feature/input' into feature/post-write-page

baegyeong committed:
Merge branch 'feature/start-post-page' into feature/input

baegyeong committed:
[style]: input 컴포넌트 margin 적용

baegyeong committed:
[feat]: select용 input 컴포넌트 구현

baegyeong committed:
[feat]: cafe, bank 데이터 추가

rktdnjs committed:
[conf] : eslint jsx-ally 속성 off & .gitignore .env 내용 추가

rktdnjs committed:
[style] : 로더 컴포넌트 적용에 필요한 CSS & 카카오 버튼 이미지 추가

rktdnjs committed:
[feat] : 카카오 소셜 로그인 페이지 개발

- 카카오 인가 코드를 받아서 처리하기까지의 큰 틀 구현
- 로그인 이후에 계좌정보 유무에 따른 추가 정보 기입 페이지 구현 필요
- 현재는 임시로 토큰을 만들어, 바로 로그인 처리 및 토큰 저장이 되도록 구현

kimdanbin committed:
[feat] : footer 기능 구현

rktdnjs committed:
Merge pull request #7 from baegyeong/feature/input

[Feat]: Input, SelectInput, Label, SubLabel 컴포넌트 구현

rktdnjs committed:
Merge branch 'weekly-4' into feature/button

rktdnjs committed:
Merge pull request #9 from rktdnjs/feature/button

[Feat]: Button 컴포넌트 구현

rktdnjs committed:
Merge pull request #8 from rktdnjs/feature/card

[Feat]: Card 컴포넌트 구현 & 유틸 함수(comma & date) 구현

rktdnjs committed:
Merge pull request #13 from kimdanbin/feature/footer

[Feat]: Footer 컴포넌트 구현

rktdnjs committed:
Merge branch 'weekly-4' into feature/login-page

rktdnjs committed:
Merge pull request #16 from rktdnjs/feature/login-page

[Feat] : 카카오 소셜 로그인 페이지 구현

rktdnjs committed:
[conf] : PR 템플릿 업데이트

- 관련 이슈 탭 추가

rktdnjs committed:
[style] : 페이지 레이아웃 box-sizing 추가 및 배경색 적용

kimdanbin committed:
[feat] : 메인페이지 Nav 구현

rktdnjs committed:
Merge pull request #17 from kimdanbin/feature/nav

[Feat]: Nav 컴포넌트 구현

rktdnjs committed:
Merge pull request #18 from rktdnjs/develop

[style] : 페이지 레이아웃 box-sizing 추가 및 배경색 적용

rktdnjs committed:
[style] : HomePage Nav 추가 및 공고 작성 탭 경로 변경 & Nav 로고 수정

rktdnjs committed:
Merge pull request #19 from rktdnjs/develop

[style] : HomePage Nav 추가 및 공고 작성 탭 경로 변경 & Nav 로고 수정

rktdnjs committed:
Merge pull request #20 from Step3-kakao-tech-campus/weekly-4

[4주차] : 개발 사항 Merge

rktdnjs committed:
Merge pull request #21 from Step3-kakao-tech-campus/develop

[4주차] : 개발 사항 Merge(develop → master)

rktdnjs committed:
[feat] : LoginNav.jsx 구현

- 로그인 페이지 상단에 사용하는 Nav 구현
- 화살표 & 홈 아이콘 클릭시 홈 페이지로 이동

rktdnjs committed:
[feat] : 매칭 여부에 따른 조건부 렌더링 구현

- 중앙 매칭완료 문구 추가 필요
- 매칭 여부에 따라 배경색, 마감기한 색상, 화살표 색상, 마우스 커서 모양 동적으로 변경

rktdnjs committed:
[conf] : tailwind-scrollbar-hide 플러그인 추가

rktdnjs committed:
[feat] : 매칭 여부에 따른 매칭완료 문구 렌더링 구현

rktdnjs committed:
[style] : Link 태그 static 속성 제거

baegyeong committed:
[fix]: 버튼 컴포넌트로 변경

- 버튼 컴포넌트 추가
- Link로 navigate 지정

baegyeong committed:
Merge branch 'develop' of https://github.com/Step3-kakao-tech-campus/Team12_FE into feature/post-write-page

baegyeong committed:
Merge branch 'feature/start-post-page' into feature/post-write-page

baegyeong committed:
[feat]: 메인, 로그인 페이지 외에서 사용하는 nav바 컴포넌트

yoeubi commented on src/utils/date.js:
```diff
@@ -0,0 +1,12 @@
+// 년월일까지 필요한 경우의 코드
+// ${myDate.getFullYear()}-${myDate.getMonth() + 1}-${myDate.getDate()}
+
+const date = (timestamp) => {
```
이름을 더 명확하게 짓는게 어떨까요?

date -> getDeadlineDate

rktdnjs replied:
해당 피드백 반영하여 수정하도록 하겠습니다! :)

yoeubi commented on src/utils/comma.js:
```diff
@@ -0,0 +1,19 @@
+    num = parseInt(num);
+  }
+
+  return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
```
```suggestion
  return num.toLocaleString()
```
이런 메소드가 있습니다.

yoeubi commented on src/components/atoms/Button.jsx:
```diff
@@ -0,0 +1,26 @@
+// 필요에 맞게 너비, 높이, 배경색, 글자색, 테두리둥금 수치를 상위 컴포넌트에서 조정
+// props default 값은 로그인 버튼 기준입니다!
+const Button = ({
+  type,
```
```suggestion
  type = 'button',
```

yoeubi commented on src/components/atoms/Button.jsx:
```diff
@@ -0,0 +1,26 @@
+      type={type || 'button'}
+      onClick={onClick}
+      disabled={disabled}
+      className={`${width} ${height} ${textColor} ${bgColor} ${bdRadius}`}
```
```suggestion
      className={[width, height, textColor, bgColor, bdRadius].join(" ")}
```

rktdnjs replied:
`join()`을 이용해서 이렇게 수정할 수도 있군요..!! 이 내용도 차후에 피드백 반영하여 수정하도록 하겠습니다! 감사합니다 :)

yoeubi reviewed (approved):
이번주도 고생하셨습니다.

아직 개발이 많이 진척이 되지는 않아서 제가 따로 피드백을 드릴게없습니다.
그리고 코드 스타일은 2단계 그대로인 이유도 있습니다.

baegyeong committed:
[feat]: Line 컴포넌트 구현

baegyeong committed:
[feat]: 동그라미 버튼 구현

baegyeong committed:
[feat]: 공고 작성 상단의 navigate 컴포넌트 구현

baegyeong committed:
[feat]: 공고 작성 하단의 navigate 컴포넌트 구현

kimdanbin committed:
[feat] : 메인페이지 구현

rktdnjs committed:
[feat] : 로그인 로직 추가 & 에러 페이지 추가

- src/arpis/index.js : axios 인스턴스 활용 코드 및 request & response interceptor 추가
- request 시에는 토큰을 헤더에 담아 백엔드로 요청
- response 시에 에러가 존재할 경우 특정 번대 에러에 대해 에러 처리하도록 함
- ErrorPage.jsx : 404 등의 에러에 대해 해당 페이지를 띄우는데 사용

rktdnjs committed:
[feat] : LoginNav 컴포넌트 뒤로가기 기능 보완

baegyeong committed:
[fix]: Input 컴포넌트 width를 prop으로 받기

baegyeong committed:
[fix]: 현재 위치에 따라 색상 다르게 지정

baegyeong committed:
[fix]: onClick 함수를 prop으로 변경

baegyeong committed:
[fix]: naviate prop에 따라 조건부 렌더링

baegyeong committed:
[feat]: 작성 페이지에 따라 컴포넌트 생성

- OrderDeadLine
- OrderInfo
- OrderRequest

baegyeong committed:
[fix]: 작성 폼에 따라 렌더링 되는 컴포넌트 분기

rktdnjs committed:
[feat] : ArticleNav 컴포넌트 구현

baegyeong committed:
[fix]: width 속성 따로 지정

kimdanbin committed:
[feat] : 상세페이지로 넘어가는 부분 수정

baegyeong committed:
[fix]: 버튼 넓이, 높이를 상수로 지정

baegyeong committed:
[fix]: 버튼 레이아웃 하단에 정렬

baegyeong committed:
[style]: 입력폼 margin 지정

baegyeong committed:
[style]: 상단 navigate margin 지정

kimdanbin committed:
[feat] : import부분 수정

kimdanbin committed:
[feat] : 공고상세페이지 구현

rktdnjs committed:
[feat] : 공고 현황 페이지 레이아웃 구성 & 공고글 렌더링 로직 구현

- PostListPage = Nav + FilterForm(필터링 기능)(미구현) + Cards(공고글 렌더링)
- Cards 컴포넌트에서는 PostListPage로부터 data를 articles prop으로 내려받음
- Card 컴포넌트에서는 Cards로부터 렌더링에 필요한 다양한 정보들을 prop으로 내려받음
- 현재 Card 컴포넌트에서 공고 상세페이지로 /articles/before/${article.boardId}의 형태를 가지나, 좀 생각해보아야함
- 사용자가 지정한 매장 이름에 따라 필터링해서 보여주는 기능은 구현 필요(현재 미구현 상태)

rktdnjs committed:
[style] : 공고 현황 페이지 좌우 스크롤 적용 및 CSS 수정

- CSS 중 whitespace-nowrap 속성을 이용하여 좌우 스크롤 적용
- global.css에서 scrollbar CSS를 수정하여 스크롤바 자체의 UI 수정

baegyeong committed:
[style]: width 등 스타일 변경

baegyeong committed:
[feat]: 여러 줄 입력을 위한 TextArea 컴포넌트 구현

baegyeong committed:
[feat]: 아이템 추가, 삭제 버튼 구현

baegyeong committed:
[feat]: plus 및 minus 버튼을 molecules 단위의 컴포넌트로 구현

baegyeong committed:
[fix]: 숫자값 max, min 값 설정

baegyeong committed:
[fix]: input 컴포넌트 추가/삭제를 위한 로직 구현

baegyeong committed:
[fix]: 요청사항 입력을 위해 TextArea 컴포넌트로 수정

baegyeong committed:
[fix]: overflow 시 스크롤 위한 속성 지정

baegyeong committed:
[fix]: 시간 input max값 변경

rktdnjs committed:
Merge pull request #23 from rktdnjs/feature/login-nav

[feat] : LoginNav 컴포넌트 구현

rktdnjs committed:
Merge pull request #25 from rktdnjs/feature/card

[feat] : Card 컴포넌트 기능 보완

rktdnjs committed:
Merge pull request #26 from baegyeong/feature/start-post-page

[fix]: 버튼 컴포넌트로 변경

rktdnjs committed:
Merge pull request #28 from kimdanbin/feature/home-page

[feat] : 메인페이지 구현

rktdnjs committed:
Merge branch 'weekly-5' into feature/login-page

rktdnjs committed:
Merge pull request #31 from rktdnjs/feature/login-page

[feat] : 로그인 로직 추가 & 에러 페이지 추가

rktdnjs committed:
Merge branch 'weekly-5' into feature/post-navigate

rktdnjs committed:
Merge pull request #32 from baegyeong/feature/post-navigate

[feat]: 공고 작성 페이지 navigate 기능 구현

rktdnjs committed:
Merge branch 'weekly-5' into feature/article-page

rktdnjs committed:
Merge pull request #33 from rktdnjs/feature/article-page

[feat] : 공고 현황 페이지 레이아웃 & 공고글 렌더링 구현

rktdnjs committed:
Merge pull request #35 from kimdanbin/feature/detail-post-page

[feat] : 공고 상세 페이지 디자인 구현

rktdnjs committed:
Merge pull request #36 from baegyeong/feature/post-write-page

[feat]: 공고작성 페이지 일부 구현

rktdnjs committed:
[refactor] : 5주차 개발 사항 코드 Merge 및 일부 리팩토링

- App.js 에서 사용되는 페이지 및 일부 컴포넌트 내 페이지 라우팅 경로 수정
ㄴ공고글과 관련된 페이지는 Post... 네이밍으로 통일
- 로그인을 위해 임시로 MyPage를 두어 로그아웃 상태에 MyPage 접속시 로그인 페이지로 유도
- KakaoOuathPage에서 로그인 진행중에 로더화면을 확인할 수 있게 2초의 Delay 지정(차후 삭제 예정)
- 공고페이지 & 마이페이지에서 사용되는 전용 Nav는 OtherNav 컴포넌트로 통합
- PostWriteIntroPage & PostDetailPage 상단에 OtherNav 컴포넌트 적용

rktdnjs committed:
Merge pull request #37 from rktdnjs/develop

[5주차] : 5주차 개발 코드 Merge 및 일부 리팩토링

rktdnjs committed:
Merge pull request #38 from Step3-kakao-tech-campus/weekly-5

[5주차] : 개발 사항 Merge(weekly-5 → develop)

rktdnjs committed:
Merge pull request #39 from Step3-kakao-tech-campus/develop

[5주차] : 개발 사항 Merge(develop → master)
